pkg/util/buffered_chan.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package util
+
+import (
+	"context"
+	"sync"
+)
+
+// BufferedChan behaves like a `chan []interface{}` (See thread safety for restrictions), but is
+// most efficient as it uses internally a channel of []interface{}.
```

Comment:
It would be good here to add more comment content on how the backing pool ties into the efficiency